Add verbose and commit-header functionality

ravibrock opened the pull request:
# Verbose
Added an option to display additional information about submissions in commit logs and folder names.

# Commit header
The prefix for automated commits is now changeable. I made some other changes to commit formatting that serve to reduce redundancy and provide more novel information about each commit while still keeping the automated messages reasonably concise.

# Folders
I also made it optional to always put problems into their own repo (so that each submission folder can be in the repo's base directory if desired, while still potentially segregating submissions into per-user repos for collaboration).

# Miscellaneous
There are also some small tweaks like adding newlines at the end of submission files - recommended for git, but not added automatically by Leetcode. Some diffs are also caused by my Vim config stripping spaces from the end of lines or adding newlines at the bottoms of code files.

# Other
The README yaml example might need updating with a new tag if this is merged. Not sure but didn't want to add a new release/tag myself.

# Images:
The screenshot below was generated by this yaml (linked to my fork for testing):
```yaml
name: Sync Leetcode

on:
  workflow_dispatch:
  schedule:
    - cron: "0 0 * * *"

jobs:
  build:
    runs-on: ubuntu-latest

    steps:
      - name: Sync
        uses: ravibrock/leetcode-sync@v1.6
        with:
          github-token: ${{ github.token }}
          leetcode-csrf-token: ${{ secrets.CSRFTOKEN }}
          leetcode-session: ${{ secrets.LEETCODE_SESSION }}
          verbose: true
          commit-header: '[LC Sync]'
```
![Screenshot 2023-08-10 at 18 07 31](https://github.com/joshcai/leetcode-sync/assets/66334356/6c02aa1c-6701-4882-af97-7f9894b5ee9b)

And the last screenshot by this one:
```yaml
name: Sync Leetcode

on:
  workflow_dispatch:
  schedule:
    - cron: "0 0 * * *"

jobs:
  build:
    runs-on: ubuntu-latest

    steps:
      - name: Sync
        uses: ravibrock/leetcode-sync@v1.6
        with:
          github-token: ${{ github.token }}
          leetcode-csrf-token: ${{ secrets.CSRFTOKEN }}
          leetcode-session: ${{ secrets.LEETCODE_SESSION }}
          destination-folder: 'problems'
          verbose: false
          commit-header: '[LC Sync]'
```
![Screenshot 2023-08-10 at 18 43 35](https://github.com/joshcai/leetcode-sync/assets/66334356/3d053aa6-47e6-4f6e-a280-720f139638e1)

joshcai commented on src/action.js:
```diff
@@ -33,8 +33,60 @@ function log(message) {
+      const questionId = pad(response.data.data.submissionDetails.question.questionId.toString());
+
+      log(`Got info for submission #${submission.id}`);
+      return { runtimeperc: runtimePercentile, memoryperc: memoryPercentile, qid: questionId };
```
nit: Could we do `runtimePerc` and `memoryPerc` for consistency with other naming?

joshcai commented on src/action.js:
```diff
@@ -33,8 +33,60 @@ function log(message) {
 
+function pad(n) {
+    var s = '000' + n;
+    return s.substring(s.length-4);
```
nit: Although unlikely to hit more than 4 digits anytime soon, maybe we can still prevent this from being an error if that ever did happen? e.g. one option would be to just return `n` as is if it's greater than 4 already

joshcai commented on README.md:
```diff
@@ -64,25 +66,27 @@ GitHub Action for automatically syncing LeetCode submissions to a GitHub reposit
 - `filter-duplicate-secs`: Number of seconds after an accepted solution to ignore other accepted solutions for the same problem, default: 86400 (1 day)
-- `destination-folder` _(optional)_: The folder in your repo to save the submissions to (necessary for shared repos)
+- `destination-folder` _(optional)_: The folder in your repo to save the submissions to (necessary for shared repos), default: _none_
+- `verbose` _(optional)_: Requires an additional API call but adds submission percentiles and question numbers to the repo, default: true
```
nit: Maybe put the description first, e.g. 

Adds submission percentiles and question numbers to the repo (requires an additional API call)

joshcai commented on src/action.js:
```diff
@@ -1,7 +1,7 @@
 const { Octokit } = require('@octokit/rest');
 
-const COMMIT_MESSAGE = 'Sync LeetCode submission';
+const COMMIT_MESSAGE = '[LeetCode Sync]';
```
Given that this isn't backwards-compatible and is now customizable, maybe we can leave it out of this PR? 

If we did want to change the default, it would probably be good to still look for the old one during a transition period.

joshcai reviewed (commented):
Thank you for the great contributions! Looks good overall except for some minor nits.

ravibrock commented:
Should be done now, thank you!

- Merged changes from @jaspinderkohli 
  - Updated `question.md` file name to `README.md` so Github will display it automatically
  - Updated `question_data` variable name to `questionData` to conform to camel case found elsewhere
- Addressed all the comments you added here

joshcai commented:
Thank you!
